### PR TITLE
Add signed up user to db & hash pw

### PR DESCRIPTION
**Pull request by wasanderes94**
- Added functionality for creating and adding a signed up user to db to AuthorizationViewModel 
- Added Bcrypt Library for hashing pw

### CODE REVIEW

File: app/src/main/java/org/feature/fox/coffee_counter/util/Event.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.feature.fox.coffee_counter.util
+
+open class Event<out T>(private val content: T) {
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/util/Event.kt:3]: This is currently unused, what are the plans for this class?

File: app/src/main/java/org/feature/fox/coffee_counter/ui/transaction/TransactionViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class TransactionViewModel @Inject constructor(
+    private val userRepository: UserRepository,
+    private val itemRepository: ItemRepository
+) : ViewModel() {
+
+    //val transactions = userRepository.getTransactions()
+    //val totalBalance = userRepository.observeTotalBalanceOfUser()
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/ui/transaction/TransactionViewModel.kt:16]: Please add a TODO or remove the comments.

File: app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt [hunk 3]
```diff
@@ -65,12 +68,23 @@ class AuthenticationViewModel @Inject constructor(
...
+            User(
+                idState.value.text,
+                nameState.value.text,
+                false,
+                BCrypt.withDefaults().hashToString(
+                    12, passwordState.value.text.toCharArray()
+                )
+            )
+        )
         val registerBody = UserBody(
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/ui/authentication/AuthenticationViewModel.kt:81]: Shouldn't this come after a successful registration?
If yes, please move it after the `if(response.data == null)` one line before `switchToLogin`.

File: app/src/main/java/org/feature/fox/coffee_counter/ui/transaction/TransactionActivity.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            CoffeeCounterTheme {
+                HistoryView()
+            }
+        }
+    }
+}
```
Comment (Morphclue) [inline app/src/main/java/org/feature/fox/coffee_counter/ui/transaction/TransactionActivity.kt:24]: We are currently using the CoreActivity for all four views. Therefore i suggest removing this activity for now.
This is due to the use of the NavigationBar that automatically leads to all 4 basic views.

**Review by Morphclue — CHANGES_REQUESTED** [review]

**Comment by wasanderes94** [comment]
Will be opened later again

**Pull request closed without merging**